### PR TITLE
fix(docs): Change docs link to Authenticator translation keys

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

Previous link: https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/i18n/translations.ts
Updated link: https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/i18n/dictionaries/authenticator/en.ts

#### Issue #, if available

Fixes: https://github.com/aws-amplify/amplify-ui/issues/3310

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/connected-components/authenticator/customization/customization.i18n.react-native.mdx [hunk 1]
```diff
@@ -23,7 +23,7 @@ These [translations](https://github.com/aws-amplify/amplify-ui/blob/main/package
 
 <Fragment>{({ platform }) => import(`./i18n/i18n.${platform}.mdx`)}</Fragment>
 
-The list of available keys are available [here](https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/i18n/translations.ts).
+The list of available keys are available [here](https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/i18n/dictionaries/authenticator/en.ts).
```
Comment (calebpollman) [inline docs/src/pages/[platform]/connected-components/authenticator/customization/customization.i18n.react-native.mdx:26]: How does making this change prevent the same issue from occurring in the future?

Comment (joebuono) [reply docs/src/pages/[platform]/connected-components/authenticator/customization/customization.i18n.react-native.mdx:26]: After chatting offline, the main concern is that linking to source code is fragile.

Comment (reesscot) [reply docs/src/pages/[platform]/connected-components/authenticator/customization/customization.i18n.react-native.mdx:26]: Long term fix could be to embed the text and to snapshot test all the docs pages? This is a potential problem with any fragment, right?

**Review by reesscot — APPROVED** [review]
LGTM

**Comment by joebuono** [comment]
Re-opened this as a short-term fix. Longer-term fix will be forthcoming.

**Review by hbuchel — APPROVED** [review]

**Pull request merged**